pkg/workflows/builder_test.go

```diff
@@ -0,0 +1,307 @@
+package workflows_test
+
+import (
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/yaml"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/targets/chainwriter"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers/streams"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/testdata/fixtures/capabilities/notstreams"
+)
+
+// Note that the set of tests in this file cover the conversion from existing YAM -> code
+// along with testing the structure of what is generated from the builders.
+// This implicitly tests the code generators functionally, as the generated code is used in the tests.
+
+type Config struct {
+	Workflow    workflows.NewWorkflowParams
+	Streams     *streams.TriggerConfig
+	Ocr         *ocr3.ConsensusConfig
+	ChainWriter *chainwriter.TargetConfig
+	TargetChain string
+}
+
+func NewWorkflowSpec(rawConfig []byte) (*workflows.WorkflowSpecFactory, error) {
+	conf, err := UnmarshalYaml[Config](rawConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	workflow := workflows.NewWorkflow(conf.Workflow)
+	streamsTrigger := conf.Streams.New(workflow)
+	consensus := conf.Ocr.New(workflow, "ccip_feeds", ocr3.ConsensusInput{
+		Observations: workflows.ListOf[streams.Feed](streamsTrigger)},
```

Comment:
Would it be possible and worthwhile to generate or define alternate names for these types?
```go
package streams
type FeedList workflows.ListOf[Feed]
```
```suggestion
		Observations: streams.FeedList(streamsTrigger)},
```
If there is a generic alias problem, those should be possible in go 1.23:
```go
package streams
type FeedList = workflows.ListOf[Feed]
```

Reply:
ListOf isn't a type it's a function call.